conformance_tests/test_tiovx/test_bidir_ext_standardnode.c

```diff
@@ -0,0 +1,220 @@
+    vxUnmapImagePatch(image, id);
+    /* On failure return a non-zero value encoding the pixels, easily readable in decimal */
+    return (aa == a) && (bb  == b) ? VX_SUCCESS : aa * 1000 + bb + 1000000;
+}
+
+/* Write some data on a 16-bit image */
+void bd_std_node_writeImageS16(vx_image image, vx_int16 a, vx_int16 b)
+{
+    vx_map_id id;
+    vx_rectangle_t rect = {.start_x = 0, .end_x = 1, .start_y = 0, .end_y = 2};
+    vx_imagepatch_addressing_t ipa;
+    void *ptr;
+    vxMapImagePatch(image, &rect, 0, &id, &ipa, &ptr, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
+    *(vx_int16 *)vxFormatImagePatchAddress2d(ptr, 0, 0, &ipa) = a;
+    *(vx_int16 *)vxFormatImagePatchAddress2d(ptr, 0, 1, &ipa) = b;
+    vxUnmapImagePatch(image, id);
+}
+
+/* Check data in an 8-bit image */
+vx_status checkImageS16(vx_image image, vx_int16 a, vx_int16 b)
+{
+    vx_map_id id;
+    vx_rectangle_t rect = {.start_x = 0, .end_x = 1, .start_y = 0, .end_y = 2};
+    vx_imagepatch_addressing_t ipa;
+    void *ptr;
+    vxMapImagePatch(image, &rect, 0, &id, &ipa, &ptr, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
+    vx_int16 aa = *(vx_int16 *)vxFormatImagePatchAddress2d(ptr, 0, 0, &ipa);
+    vx_int16 bb = *(vx_int16 *)vxFormatImagePatchAddress2d(ptr, 0, 1, &ipa);
+    vxUnmapImagePatch(image, id);
+    /* On failure output the data and return VX_FAILURE */
+    if (aa == a && bb == b)
+        return VX_SUCCESS;
+    return VX_FAILURE;
+}
+
+vx_status releaseNode(vx_node node)
+{
+    return vxReleaseNode(&node);
+}
+
+void testAccumulate(vx_context context)
+{
+    vx_graph graph;
+    vx_image in_param, accum_param;
+
+    ASSERT_VX_OBJECT(graph = vxCreateGraph(context), VX_TYPE_GRAPH);
+    ASSERT_VX_OBJECT(in_param = vxCreateImage(context, 8, 8, VX_DF_IMAGE_U8), VX_TYPE_IMAGE);
+    ASSERT_VX_OBJECT(accum_param = vxCreateImage(context, 8, 8, VX_DF_IMAGE_S16), VX_TYPE_IMAGE);
+    bd_std_node_writeImage(in_param, 0x01, 0xFF);
+    bd_std_node_writeImageS16(accum_param, 0x0100, 0x7F03);
+    VX_CALL(releaseNode(vxAccumulateImageNode(graph, in_param, accum_param)));
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxProcessGraph(graph));
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, checkImageS16(accum_param, 0x0101, 0x7FFF));
+    VX_CALL(vxReleaseImage(&in_param));
+    VX_CALL(vxReleaseImage(&accum_param));
+    VX_CALL(vxReleaseGraph(&graph));
+}
+
+void testAccumulateBad(vx_context context)
+{
+    vx_graph graph;
+    vx_image in_param, accum_param;
+
+    ASSERT_VX_OBJECT(graph = vxCreateGraph(context), VX_TYPE_GRAPH);
+    ASSERT_VX_OBJECT(in_param = vxCreateImage(context, 8, 8, VX_DF_IMAGE_U8), VX_TYPE_IMAGE);
+    ASSERT_VX_OBJECT(accum_param  = vxCreateImage(context, 8, 8, VX_DF_IMAGE_U8), VX_TYPE_IMAGE);
+    bd_std_node_writeImage(in_param, 0x01, 0xFF);
+    bd_std_node_writeImage(accum_param, 0xFF, 0x03);
+    vx_node node = vxAccumulateImageNode(graph, NULL, accum_param);
+    EXPECT_NE_VX_STATUS(VX_SUCCESS, vxGetStatus(node));
```

Comment:
The remaining errors are listed below and should be fairly straightforward to fix

/home/dockeruser/tiovx_kw_workspace/workarea/tiovx/conformance_tests/test_tiovx/test_bidir_ext_standardnode.c:118:49: error: incompatible pointer types passing 'vx_node' (aka 'struct _vx_node *') to parameter of type 'vx_reference' (aka 'struct _vx_reference *') [-Werror,-Wincompatible-pointer-types]
    EXPECT_NE_VX_STATUS(VX_SUCCESS, vxGetStatus(node));
                                                ^~~~
/home/dockeruser/tiovx_kw_workspace/workarea/tiovx/conformance_tests/test_engine/test_utils.h:104:36: note: expanded from macro 'EXPECT_NE_VX_STATUS'
        vx_status s1 = (vx_status)(val2);                               \
                                   ^~~~
/home/dockeruser/tiovx_kw_workspace/workarea/tiovx/include/VX/vx_api.h:142:61: note: passing argument to parameter 'reference' here
VX_API_ENTRY vx_status VX_API_CALL vxGetStatus(vx_reference reference);
                                                            ^
/home/dockeruser/tiovx_kw_workspace/workarea/tiovx/conformance_tests/test_tiovx/test_bidir_ext_standardnode.c:155:49: error: incompatible pointer types passing 'vx_node' (aka 'struct _vx_node *') to parameter of type 'vx_reference' (aka 'struct _vx_reference *') [-Werror,-Wincompatible-pointer-types]
    EXPECT_NE_VX_STATUS(VX_SUCCESS, vxGetStatus(node));
                                                ^~~~
/home/dockeruser/tiovx_kw_workspace/workarea/tiovx/conformance_tests/test_engine/test_utils.h:104:36: note: expanded from macro 'EXPECT_NE_VX_STATUS'
        vx_status s1 = (vx_status)(val2);                               \
                                   ^~~~
/home/dockeruser/tiovx_kw_workspace/workarea/tiovx/include/VX/vx_api.h:142:61: note: passing argument to parameter 'reference' here
VX_API_ENTRY vx_status VX_API_CALL vxGetStatus(vx_reference reference);
                                                            ^
/home/dockeruser/tiovx_kw_workspace/workarea/tiovx/conformance_tests/test_tiovx/test_bidir_ext_standardnode.c:194:49: error: incompatible pointer types passing 'vx_node' (aka 'struct _vx_node *') to parameter of type 'vx_reference' (aka 'struct _vx_reference *') [-Werror,-Wincompatible-pointer-types]
    EXPECT_NE_VX_STATUS(VX_SUCCESS, vxGetStatus(node));
                                                ^~~~
/home/dockeruser/tiovx_kw_workspace/workarea/tiovx/conformance_tests/test_engine/test_utils.h:104:36: note: expanded from macro 'EXPECT_NE_VX_STATUS'
        vx_status s1 = (vx_status)(val2);                               \
                                   ^~~~
/home/dockeruser/tiovx_kw_workspace/workarea/tiovx/include/VX/vx_api.h:142:61: note: passing argument to parameter 'reference' here
VX_API_ENTRY vx_status VX_API_CALL vxGetStatus(vx_reference reference);

Reply:
done

Reply:
I am still seeing a number of build errors even with this set of changes in the attached log.  Can you please take a look?

[build_errors.txt](https://github.com/TexasInstruments/tiovx/files/13421363/build_errors.txt)

Reply:
I found the problem, you cannot use the VX_CALL or ASSERT_EQ_VX_STATUS inside a function which return a vx_status, because of the return CT_DO_FAIL from the macro itself.

So I remove these call in the functions, please have a look. I cannot verify myself if this is fine, as we do not have our MISRA check switched on for the OVX repo.

Reply:
Thanks.  However, I am still seeing build errors as per the attached log.  Can you please fix?
[build_error_11_30.txt](https://github.com/TexasInstruments/tiovx/files/13515151/build_error_11_30.txt)

Reply:
I looked on it, not clear why there is an error here. I synced with the official header from Khronos, let me know the result.,